File: practice/algorithms/warmup/time_conversion/TimeConversion.js
function processData(input) {
    var hours = parseInt(input.substring(0, input.indexOf(':')));
    var minutes = parseInt(input.substring(input.indexOf(':') + 1, input.lastIndexOf(':')));
    var seconds = parseInt(input.substring(input.lastIndexOf(':') + 1, input.lastIndexOf(':') + 3));
    var newHours;
    if(hours === 12) {
        newHours = 0;
    } else {
        newHours = hours;
    }
    var shift;
    if(input.toLowerCase().indexOf('pm') > -1) {
        shift = 12;
    } else {
        shift = 0;
    }

    var formatedHours = format((newHours + shift).toString());
    var formatedMinutes = format(minutes.toString());
    var formatedSeconds = format(seconds.toString());

    console.log(formatedHours + ":" + formatedMinutes + ":" + formatedSeconds);

    function format(numString) {
        if(numString.length === 1) {
            return "0" + numString;
        }
        return numString;
    }
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
    processData(_input);
});

File: practice/algorithms/warmup/time_conversion/time_conversion.rb
input_time = gets.to_s
hours = input_time.slice(0, 2).to_i
minutes = input_time.slice(input_time.index(":") + 1, 2).to_i
seconds = input_time.slice(input_time.index(":", input_time.index(":") + 1) + 1, 2).to_i

newHours = hours
if hours == 12 then
  newHours = 0 
end

shift = 0
if input_time.upcase.include? "PM" then
  shift = 12
end

puts "%02d:%02d:%02d" % [newHours + shift, minutes, seconds]
